controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -494,10 +496,35 @@ func (r *BareMetalHostReconciler) actionDeleting(prov provisioner.Provisioner, i
 		return deleteComplete{}
 	}
 
+	// Only try to power off `maxPowerOffRetryCount` times before giving up, and deleting the host anyway
+	if info.host.Status.ErrorType != metal3v1alpha1.PowerManagementError || (info.host.Status.ErrorType == metal3v1alpha1.PowerManagementError && info.host.Status.ErrorCount < maxPowerOffRetryCount) {
+		info.log.Info("host ready to be powered off")
+		provResult, err := prov.PowerOff(
+			metal3v1alpha1.RebootModeHard,
+			info.host.Status.ErrorType == metal3v1alpha1.PowerManagementError)
+
+		if err != nil {
+			return actionError{errors.Wrap(err, "failed to power off")}
```

Comment:
I just noticed that we have some code in `handleDeprovisioning()` that ensures we don't get stuck in an error loop in the case where the host is not registered in Ironic and we don't actually have credentials to register it: https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/host_state_machine.go#L523-L528

We'll need similar code here, because failures are subject to the `maxPowerOffRetryCount` but errors will retry forever unless we short-circuit them.

An alternative would be to not attempt to power off if the `ErrorType` going in is a Deprovisioning error. I'm not certain that this would be the right thing though, because there could conceivably be cases where deprovisioning is going to fail (e.g. host is borked so it won't boot the IPA image) but we can still turn the power off (BMC still works).